test/extended/util/cli.go

```diff
@@ -0,0 +1,219 @@
+		FatalErr(fmt.Errorf("The configPath can't be empty"))
+	}
+	client.configPath = configPath
+	return client.SetNamespace(project)
+}
+
+// NewGinkoCLI initialize the upstream E2E framework and setup project role bindings.
+// This function also initialize the Ginko BeforeEach hook to setup the role bindings
+// for the namespace created by the E2E framework.
+func NewGinkoCLI(project, configPath string) *CLI {
+	// The namespace will be set by E2E framework.
+	client := NewCLI(project, configPath).SetNamespace("")
+	BeforeEach(func() {
+		if len(client.Namespace()) > 0 {
+			Logf("Adding project role bindings to %q namespace", client.Namespace())
+			client.SetupRoleBindings()
+		} else {
+			Failf("Framework does not have the namespace set")
+		}
+	})
+	return client
+}
+
+// OsFramework returns OpenShift framework
+func (c *CLI) OsFramework() *OsFramework {
+	return NewOsFramework(c.kubeFramework.Namespace, c.AdminRESTClient())
+}
+
+// KubeFramework returns Kubernetes framework
+func (c *CLI) KubeFramework() *Framework {
+	return c.kubeFramework
+}
+
+// SetNamespace overrides the namespace set by framework
+func (c *CLI) SetNamespace(ns string) *CLI {
+	if c.kubeFramework == nil {
+		FatalErr(fmt.Errorf("The E2E framework must be initialized"))
+	}
+	c.kubeFramework.Namespace = &kapi.Namespace{ObjectMeta: kapi.ObjectMeta{Name: ns}}
+	return c
+}
+
+// SetupRoleBindings setups a project role binding for the current namespace
+func (c *CLI) SetupRoleBindings() *CLI {
+	for _, binding := range bootstrappolicy.GetBootstrapServiceAccountProjectRoleBindings(c.Namespace()) {
+		addRole := &policy.RoleModificationOptions{
+			RoleName:            binding.RoleRef.Name,
+			RoleNamespace:       binding.RoleRef.Namespace,
+			RoleBindingAccessor: policy.NewLocalRoleBindingAccessor(c.Namespace(), c.AdminRESTClient()),
+			Users:               binding.Users.List(),
+			Groups:              binding.Groups.List(),
+		}
+		if err := addRole.AddRole(); err != nil {
+			Failf("Unable to add role binding %+v to namespace %q: %v", addRole, c.Namespace(), err)
+		}
+	}
+	return c
+}
+
+// Verbose turns on verbose messages for current command
+func (c *CLI) Verbose() *CLI {
+	c.verbose = true
+	return c
+}
+
+// AdminRESTClient return the current project namespace REST client
+func (c *CLI) AdminRESTClient() *client.Client {
+	client, err := testutil.GetClusterAdminClient(c.configPath)
+	if err != nil {
+		FatalErr(err)
+	}
+	return client
+}
+
+// AdminKubeRESTClient returns the current project namespace Kubernetes client
+func (c *CLI) AdminKubeRESTClient() *kclient.Client {
+	if c.kubeFramework.Client != nil {
+		return c.kubeFramework.Client
+	}
+	client, err := testutil.GetClusterAdminKubeClient(c.configPath)
+	if err != nil {
+		FatalErr(err)
+	}
+	return client
+}
+
+// Namespace returns the name of the namespace used in the current test case
+func (c *CLI) Namespace() string {
+	if c.kubeFramework == nil || c.kubeFramework.Namespace == nil {
```

Comment:
You don't make this nil check for `kubeFramework` elsewhere.  If we require it, force it during construction.  Otherwise, check it everywhere.  Probably easier to require it.